secboot/secboot.go

```diff
@@ -251,3 +253,50 @@ func MarkSuccessful() error {
 
 	return nil
 }
+
+// AuthMode corresponds to an authentication mechanism.
+type AuthMode string
+
+const (
+	AuthModePassphrase AuthMode = "passphrase"
+	// TODO: Add PIN option when secboot support lands.
+)
+
+// VolumesAuthOptions contains options for the volumes authentication
+// mechanism (e.g. passphrase authentication).
+type VolumesAuthOptions struct {
+	Mode       AuthMode `json:"mode,omitempty"`
+	Passphrase string   `json:"passphrase,omitempty"`
+	// TODO: Add PIN option when secboot support lands.
+	KDFType string        `json:"kdf-type,omitempty"`
+	KDFTime time.Duration `json:"kdf-time,omitempty"`
+}
+
+// Validates authentication options.
+func (o *VolumesAuthOptions) Validate() error {
+	if o == nil {
+		return nil
+	}
+
+	switch o.Mode {
+	case AuthModePassphrase:
+		// TODO: Add entropy/quality checks on passphrase.
+		if len(o.Passphrase) == 0 {
+			return fmt.Errorf("passphrase cannot be empty")
+		}
+	default:
+		return fmt.Errorf("invalid authentication mode %q, only %q is supported", o.Mode, AuthModePassphrase)
```

Comment:
supported modes will likely become a list in the next PR, so you may as well make one now

Reply:
There will only be two modes, passphrases now and PINs later. Is it worth building errors from a list?

I thinking maybe something like this
```go
	switch o.Mode {
	case AuthModePassphrase:
		// TODO: Add entropy/quality checks on passphrase.
		if len(o.Passphrase) == 0 {
			return fmt.Errorf("passphrase cannot be empty")
		}
	case AuthModePIN:
		return fmt.Errorf("%q authentication mode is not implemented", AuthModePIN)
	default:
		return fmt.Errorf("invalid authentication mode %q, only %q and %q modes are supported", o.Mode, AuthModePassphrase, AuthModePIN)
	}
```

Reply:
looks good